src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -1207,10 +1206,80 @@ public ImmutableArray<SymbolDisplayPart> ToDisplayParts(SymbolDisplayFormat form
             }
         }
 
-        internal void ReportExplicitUseOfNullabilityAttribute(in DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments, AttributeDescription attributeDescription)
+        [Flags]
+        internal enum ReservedAttributes
         {
-            // Attribute should not be set explicitly.
-            arguments.Diagnostics.Add(ErrorCode.ERR_ExplicitReservedAttr, arguments.AttributeSyntaxOpt.Location, attributeDescription.FullName);
+            DynamicAttribute = 1 << 1,
+            IsReadOnlyAttribute = 1 << 2,
+            IsUnmanagedAttribute = 1 << 3,
+            IsByRefLikeAttribute = 1 << 4,
+            TupleElementNamesAttribute = 1 << 5,
+            NullableAttribute = 1 << 6,
+            NullableContextAttribute = 1 << 7,
+            NullablePublicOnlyAttribute = 1 << 8,
+            NativeIntegerAttribute = 1 << 9,
+        }
+
+        internal bool ReportExplicitUseOfReservedAttributes(ref DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments, ReservedAttributes reserved)
```

Comment:
Seems like the members accessed on 'arguments' here are all readonly, so an `in` ref could be used

Reply:
Good catch, thanks.

---
In reply to: [373271430](https://github.com/dotnet/roslyn/pull/41160#discussion_r373271430) [](ancestors = 373271430)